tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -77,6 +81,8 @@ private static Trace createTrace(Optional<Boolean> isObservable, String traceId)
      * #setSampler configured sampler} returns true.
      */
     public static void initTrace(Optional<Boolean> isObservable, String traceId) {
+        checkForRemotingTracerCompatibility(REMOTING_TRACER_VER);
```

Comment:
Why don't we do this statically, on class load?
Might as well fail immediately if we have an incompatible classpath